components/frontend/src/utils.js

```diff
@@ -32,26 +32,74 @@ export function getMetricUnit(metric, dataModel) {
     return formatMetricUnit(metricType, metric)
 }
 
-export function getMetricDeadline(metric, report) {
+export function getMetricResponseDeadline(metric, report) {
     let deadline;
     if (metric.status === "debt_target_met" && metric.debt_end_date) {
         deadline = new Date(metric.debt_end_date)
     } else {
         const statusStart = metric.status_start || "3000-01-01"
-        const status = metric.status
-        const desiredResponseTime = report?.desired_response_times?.[status] ?? (metricReactionDeadline[status] ?? metricReactionDeadline["unknown"])
         deadline = new Date(statusStart)
-        deadline.setDate(deadline.getDate() + desiredResponseTime)
+        deadline.setDate(deadline.getDate() + getMetricDesiredResponseTime(report, metric.status))
     }
     return deadline
 }
 
-export function getMetricTimeLeft(metric, report) {
-    const deadline = getMetricDeadline(metric, report)
+export function getMetricResponseTimeLeft(metric, report) {
+    const deadline = getMetricResponseDeadline(metric, report)
     const now = new Date()
     return deadline.getTime() - now.getTime()
 }
 
+export function getMetricResponseOverruns(metric_uuid, metric, report, measurements) {
+    const scale = metric?.scale ?? "count"
+    let previousStatus;
+    const consolidatedMeasurements = [];
+    const filteredMeasurements = measurements.filter((measurement) => measurement.metric_uuid === metric_uuid)
+    filteredMeasurements.forEach((measurement) => {
+        const status = measurement?.[scale]?.status || "unknown"
+        if (status === previousStatus) {
+            consolidatedMeasurements.at(-1).end = measurement.end  // Status unchanged so merge this measurement with the previous one
+        } else {
+            consolidatedMeasurements.push(measurement);  // Status changed or first one, so keep this measurement
+        }
+        previousStatus = status
+    })
+    return consolidatedMeasurements
+}
+
+export function getMetricResponseOverrun(metric_uuid, metric, report, measurements) {
+    const consolidatedMeasurements = getMetricResponseOverruns(metric_uuid, metric, report, measurements)
+    const scale = metric?.scale ?? "count"
+    let totalOverrun = 0;  // Amount of time the desired response time was not achieved for this metric
+    const overruns = []
+    consolidatedMeasurements.forEach((measurement) => {
+        const status = measurement?.[scale]?.status || "unknown"
+        if (status in metricReactionDeadline) {
+            const desiredResponseTime = getMetricDesiredResponseTime(report, status) * 24 * 60 * 60 * 1000
```

Comment:
Does it make sense to also extract `* 24 * 60 * 60 * 1000` to a separate function? :P

Reply:
Constant you mean? Yeah, will do.